components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java

```diff
@@ -681,6 +683,11 @@ public static String getUserConsentURL(OAuth2Parameters params, String loggedInU
         String sessionDataKeyConsent = UUID.randomUUID().toString();
         try {
             if (entry != null && entry.getQueryString() != null) {
+
+                if (entry.getQueryString().contains(REQUEST_URI) && params != null) {
+                    entry.setQueryString(entry.getQueryString() +
+                            "&" + PROP_REDIRECT_URI + "=" + params.getRedirectURI());
+                }
```

Comment:
Why is this needed? Also if `PROP_REDIRECT_URI` is required downstream, then what would happen if the App is configured to skip consent?(IIRC then this code block will not execute)